### PR TITLE
Implement edited flag in room message

### PR DESCRIPTION
**Pull request by bitfriend**
- We need the flag that shows whether a message is the edited version of other.
- Edited message should show the latest body, not original body.
- Timeline-based integration test should be implemented.

### CODE REVIEW

File: native/acter/src/api/room.rs [hunk 1]
```diff
@@ -1065,7 +1065,7 @@ impl Room {
                     event_id.to_owned(),
                     MessageType::text_plain(new_msg.to_string()).into(),
                 );
-                let mut edited_content = RoomMessageEventContent::text_markdown(new_msg);
+                let mut edited_content = RoomMessageEventContent::text_plain(new_msg);
```
Comment (gnunicorn) [inline native/acter/src/api/room.rs:1068]: why though?

Comment (bitfriend) [reply native/acter/src/api/room.rs:1068]: Because here is `edit_plain_message()`.

File: native/test/src/tests/msg_edit.rs [hunk 2]
```diff
@@ -1,41 +1,85 @@
...
+    pin_mut!(stream);
+
+    let event_id = convo.send_plain_message("Hi, everyone".to_string()).await?;
     println!("event id: {event_id:?}");
 
-    let edited_id = space
+    let edited_id = convo
         .edit_plain_message(event_id.to_string(), "This is message edition".to_string())
         .await?;
+    println!("edited id: {edited_id:?}");
```
Comment (gnunicorn) [inline native/test/src/tests/msg_edit.rs:26]:
```suggestion
    info!("edited id: {edited_id:?}");
```

File: native/test/src/tests/msg_edit.rs [hunk 3]
```diff
@@ -1,41 +1,85 @@
...
-
-    let Ok(AnyTimelineEvent::MessageLike(AnyMessageLikeEvent::RoomMessage(MessageLikeEvent::Original(r)))) = ev.event.deserialize() else {
-        bail!("not the proper room event")
-    };
-
-    let msg = RoomMessage::room_message_from_event(r, space.deref().deref().clone(), false);
-    let Some(item) = msg.event_item() else {
-        bail!("This item should be event item not virtual item")
-    };
+    while let Some(diff) = stream.next().await {
```
Comment (gnunicorn) [inline native/test/src/tests/msg_edit.rs:28]: Assuming the check doesn't apply (and thus break) this tests will just idle forever asking for a next one that never comes, won't it?

How about changing this into a retry-loop with a definite end-time and failure after a few attempts instead?

File: native/test/src/tests/msg_edit.rs [hunk 4]
```diff
@@ -1,41 +1,85 @@
...
+        match diff.action().as_str() {
+            "Append" => {}
+            "Insert" => {}
+            "Set" => {
+                info!("diff index: {:?}", diff.index());
+                info!("diff value: {:?}", diff.value());
+                info!("diff values: {:?}", diff.values());
+                let Some(value) = diff.value() else {
```
Comment (gnunicorn) [inline native/test/src/tests/msg_edit.rs:36]: within tests you can just do `.expect()` ... (which is the same as `else { bail!(...` effectively and shorter to read and write) ...

File: native/test/src/utils.rs [hunk 5]
```diff
@@ -1,13 +1,36 @@
...
-    CreateSpaceSettingsBuilder, SyncState,
+    CreateConvoSettingsBuilder, CreateSpaceSettingsBuilder, SyncState,
 };
 use acter_core::templates::Engine;
 use anyhow::Result;
 use futures::{pin_mut, stream::StreamExt};
 use tracing::trace;
 use uuid::Uuid;
 
+pub async fn random_user_with_random_convo(prefix: &str) -> Result<(Client, OwnedRoomId)> {
```
Comment (gnunicorn) [inline native/test/src/utils.rs:11]: 👍

File: native/test/src/tests/msg_edit.rs [hunk 6]
```diff
@@ -1,41 +1,85 @@
...
+                assert_eq!(text_desc.body(), "This is message edition");
+                break;
+            }
+            "Remove" => {}
+            "PushBack" => {}
+            "PushFront" => {}
+            "PopBack" => {}
+            "PopFront" => {}
+            "Clear" => {}
+            "Reset" => {
```
Comment (gnunicorn) [inline native/test/src/tests/msg_edit.rs:57]: I am not sure I understand ... we expect either Reset or set? Don't we surely know which one?

**Review by gnunicorn — APPROVED** [review]
yay, tests!

Just some minor remarks about the details.

File: native/test/src/tests/msg_edit.rs (hunk 2 above)
Comment (bitfriend) [reply native/test/src/tests/msg_edit.rs]: Fixed by https://github.com/acterglobal/a3/pull/1126/commits/455a87e9276216c462259b1e1137f58970e98207

File: native/test/src/tests/msg_edit.rs (hunk 4 above)
Comment (bitfriend) [reply native/test/src/tests/msg_edit.rs]: Fixed by https://github.com/acterglobal/a3/pull/1126/commits/ee1a9f2225cad25fe1bdc8c465be3121067f40c1

File: native/test/src/tests/msg_edit.rs (hunk 6 above)
Comment (bitfriend) [reply native/test/src/tests/msg_edit.rs]: We get `Hi, everyone` msg under `PushBack` action in usual case.
We may get that msg under `Reset` action, when calling of `now_or_never()` loop was too late and timeline already arranged `Hi, everyone` into stream.
We get `This is message edition` msg under `Set` action.
As we know, `Set` action means by record modification.

**Pull request merged**